docs/pages/guides/tree-shaking.mdx

```diff
@@ -300,3 +302,211 @@ const { View, Image } = require('react-native-web');
+
+If the star export pulls in ambiguous exports such as `module.exports.ArrowUp` or `exports.ArrowDown`, then the optimization pass will not expand the star export and no exports will be removed from the barrel file. You can refer to Expo Atlas to inspect the expanded exports.
+
+You can use this with libraries like `lucide-react` to remove all icons that are not used in your app.
+
+## Recursive optimizations
+
+> Experimentally available in SDK 52 and greater
+
+Expo optimizes a module by recursing through the graph exhaustively to find unused imports. Consider the following:
+
+```js Input
+export function foo() {
+  // Because bar is used here, it cannot be removed.
+  bar();
+}
+
+export function bar() {}
+```
+
+In this case, `bar` is used in `foo`, so it cannot be removed. However, if `foo` is not used anywhere in the app, then `foo` will be removed and the module will be scanned again to see if `bar` can be removed. This process recurses 5 times for a given module before bailing out due to performance reasons.
+
+## Side-effects
+
+Expo CLI respects module side-effects according to the [Webpack system](https://webpack.js.org/guides/tree-shaking/#mark-the-file-as-side-effect-free). Side-effects are generally used for defining global variables (`console.log`) or modifying prototypes (avoid doing this).
+
+You can mark if your module has side-effects in the `package.json`:
+
+```json package.json
+{
+  "name": "library",
+  "sideEffects": ["./src/*.js"]
+}
+```
+
+Side-effects will prevent the removal of unused modules and disable module inlining to ensure JS code runs in the expected order. Side-effects will be removed if they're empty or contain only comments and directives (`"use strict"`, `"use client"`, etc).
+
+When Expo tree shaking is enabled, you can safely enable `inlineRequires` in your `metro.config.js` for production bundles. This will lazily load modules when they're evaluated, leading to faster startup time. Avoid using this feature without Expo tree shaking as it will move modules around in ways that can change the execution order of side-effects.
+
+```js metro.config.js
+const { getDefaultConfig } = require('expo/metro-config');
+
+const config = getDefaultConfig(__dirname);
+
+config.transformer.getTransformOptions = async () => ({
+  transform: {
+    experimentalImportSupport: true,
+    inlineRequires: true,
+  },
+});
+
+module.exports = config;
+```
+
+## Optimizing for tree shaking
+
+Before Expo tree shaking, React Native libraries would remove imports by wrapping them in conditional blocks such as:
+
+```js
+if (process.env.NODE_ENV === 'development') {
+  require('./dev-only').doSomething();
+}
+```
+
+This is problematic because you don't have accurate TypeScript support and it makes the graph ambiguous since you cannot statically analyze the code.
+
+With Expo tree shaking enabled, you can restructure this code to use ESM imports:
```

Comment:
```suggestion
This is problematic because you don't have accurate TypeScript support and it makes the graph ambiguous since you cannot statically analyze the code. With Expo tree shaking enabled, you can restructure this code to use ESM imports:
```

This can be paragraph.